src/zenml/integrations/gcp/services/vertex_deployment.py

```diff
@@ -0,0 +1,371 @@
+        try:
+            logger.info(
+                f"Checking status of Vertex AI Inference Endpoint: {self.status.endpoint.endpoint_name}"
+            )
+            endpoint = aiplatform.Endpoint(
+                endpoint_name=self.status.endpoint.endpoint_name,
+                location=self.config.location,
+            )
+
+            # Get detailed deployment status
+            deployment = None
+            if self.status.endpoint.deployed_model_id:
+                deployments = [
+                    d
+                    for d in endpoint.list_models()
+                    if d.model == self.status.endpoint.deployed_model_id
+                ]
+                if deployments:
+                    deployment = deployments[0]
+                    logger.info(
+                        f"Model {self.status.endpoint.deployed_model_id} was deployed to the endpoint"
+                    )
+
+            if not deployment:
+                logger.warning(
+                    "No matching deployment found, endpoint may be inactive or failed to deploy"
+                )
+                return ServiceState.INACTIVE, "No matching deployment found"
+
+            return ServiceState.ACTIVE, "Deployment is ready"
+
+        except exceptions.NotFound:
+            return ServiceState.INACTIVE, "Endpoint not found"
+        except Exception as e:
+            return ServiceState.ERROR, f"Error checking status: {str(e)}"
+
+    def predict(self, instances: List[Any]) -> List[Any]:
+        """Make a prediction using the service."""
+        if not self.is_running:
+            raise Exception(
+                "Vertex AI endpoint inference service is not running. "
+                "Please start the service before making predictions."
+            )
+
+        if not self.status.endpoint:
+            raise Exception("Endpoint information is missing.")
+
+        try:
+            endpoint = aiplatform.Endpoint(
+                endpoint_name=self.status.endpoint.endpoint_name,
+                location=self.config.location,
+            )
+
+            # Add proper prediction parameters and handle sync/async
+            predictions = endpoint.predict(
+                instances=instances,
+                deployed_model_id=self.status.endpoint.deployed_model_id.split(
+                    "/"
+                )[-1],
+                timeout=30,  # Add reasonable timeout
+            )
+
+            if not predictions:
+                raise RuntimeError("No predictions returned")
+
+        except Exception as e:
+            logger.error(f"Prediction failed: {e}")
+            raise RuntimeError(f"Prediction failed: {str(e)}")
+
+        return [predictions]
+
+    def get_logs(
+        self, follow: bool = False, tail: Optional[int] = None
+    ) -> Generator[str, bool, None]:
+        """Retrieve the service logs."""
+        # Note: Could be enhanced to actually fetch logs from Cloud Logging
+        logger.info(
```

Comment:
Can't we at least build the URL to the logs in cloud watch, which might be easier than actually fetching them